File: java-samples/Features/contractsdk-recordplayers/workflows/src/main/java/net/corda/samples/contractsdk/flows/IssueRecordPlayerFlowResponder.java
package net.corda.samples.contractsdk.flows;

import co.paralleluniverse.fibers.Suspendable;
import net.corda.core.flows.*;
import net.corda.core.transactions.SignedTransaction;

// ******************
// * Responder flow *
// ******************
@InitiatedBy(IssueRecordPlayerFlow.class)
public class IssueRecordPlayerFlowResponder extends FlowLogic<Void> {

    //private variable
    private FlowSession counterpartySession;

    //Constructor
    public IssueRecordPlayerFlowResponder(FlowSession counterpartySession) {
        this.counterpartySession = counterpartySession;
    }

    @Suspendable
    @Override
    public Void call() throws FlowException {
        SignedTransaction signedTransaction = subFlow(new SignTransactionFlow(counterpartySession) {
            @Suspendable
            @Override
            protected void checkTransaction(SignedTransaction stx) throws FlowException {
                // we could include additional checks here but it's not necessary for the purposes of the contract sdk
            }
        });

        //Stored the transaction into data base.
        subFlow(new ReceiveFinalityFlow(counterpartySession, signedTransaction.getId()));
        return null;
    }
}

File: kotlin-samples/Features/contractsdk-recordplayers/workflows/src/main/kotlin/net/corda/samples/contractsdk/flows/IssueRecordPlayerFlowResponder.kt
package net.corda.samples.contractsdk.flows

import co.paralleluniverse.fibers.Suspendable
import net.corda.core.flows.*
import net.corda.core.transactions.SignedTransaction

// ******************
// * Responder flow *
// ******************
@InitiatedBy(IssueRecordPlayerFlow::class)
class IssueRecordPlayerFlowResponder(private val counterpartySession: FlowSession) : FlowLogic<SignedTransaction>() {
    @Suspendable
    @Throws(FlowException::class)
    override fun call(): SignedTransaction {
        val signedTransaction = subFlow(object : SignTransactionFlow(counterpartySession) {
            @Suspendable
            @Throws(FlowException::class)
            override fun checkTransaction(stx: SignedTransaction) {
                // we could include additional checks here but it's not necessary for the purposes of the contract sdk
            }
        })

        //Stored the transaction into data base.
        return subFlow(ReceiveFinalityFlow(counterpartySession, signedTransaction.id))
    }
}
